message_processor.go

```diff
@@ -27,11 +31,53 @@ func (mg messageProcessor) process(lf *logPair) {
 	consumeLog(lf)
 }
 
+type tmpBuffer struct {
+	tBuf                      bytes.Buffer
+	bufferTimer               time.Time
+	bufferReset               bool
+}
+
+func (b *tmpBuffer) appendToTempBuffer(l *logdriver.LogEntry) bool {
+	// Add msg to temp buffer and disable buffer reset flag
+	ps, err := b.tBuf.Write(l.Line)
+	b.bufferReset = false
+	if err != nil {
+		logrus.WithError(err).WithField("Appending to Temp Buffer with size:", ps).Error(
+			"Error appending to temp buffer")
+		b.resetTempBuffer()
+		return false
+	}
+	return true
+}
+
+func (b *tmpBuffer) resetTempBuffer() {
+	logrus.Debug("Resetting temp buffer")
+	b.tBuf.Reset()
+	b.bufferTimer = time.Now()
+}
+
+func (b *tmpBuffer) isTempBufferHoldDurationExpired (t time.Time) bool {
```

Comment:
Please get rid of "TempBuffer" in the name since we already operated on a `tmpBuffer` object

Reply:
Same for  all these member functions

Reply:
Done.